photon-client/src/main/scala/com/linkedin/photon/ml/hyperparameter/HyperparameterHelper.scala

```diff
@@ -0,0 +1,183 @@
+    }
+
+    priorData.map { case (params, evalValue) =>
+      (DenseVector(params.toArray), evalValue)
+    }
+  }
+
+  /**
+   * Read in the JSON config file and returns the parsed hyper-parameter configurations.
+   *
+   * @param jsonConfig The config file in json format.
+   * @return A tuple containing the hyperparameter tuning mode, list of parameters, ranges of min and max for each
+   *         parameter.
+   */
+  def configFromJson(jsonConfig: String): HyperparameterConfig = {
+
+    val (tuningMode, paramDetails) = JSON.parseFull(jsonConfig) match {
+      case Some(inputConfig: Map[String, Any]) =>
+
+        val mode = inputConfig("tuning_mode") match {
+          case BAYESIAN_MODEL => HyperparameterTuningMode.BAYESIAN
+          case RANDOM_MODEL => HyperparameterTuningMode.RANDOM
+          case _ => HyperparameterTuningMode.NONE
+        }
+
+        val hyperparameterDetails = inputConfig("variables") match {
+
+          case variables: Map[String, Any] =>
+
+            variables.map {
+              case (key: String, value: Map[String, Any]) =>
+
+                (value("type"), value("min"), value("max"), value.get("transform")) match {
+                  case (varType: String, min: Double, max: Double, transform: Option[_]) =>
+                    (key, varType, min, max, transform)
+
+                  case _ => throw new IllegalArgumentException("The minimum and maximum values must be numeric")
+                }
+
+              case _ =>
+                throw new IllegalArgumentException("Each hyper-parameter configuration must be a map")
+            }
+
+          case _ =>
+            throw new IllegalArgumentException("The hyper-parameter configurations must be a map")
+        }
+
+        (mode, hyperparameterDetails)
+
+      case _ =>
+        throw new IllegalArgumentException("JSON config is not a Map[String, Any]")
+    }
+
+    val hyperparameters = paramDetails.map(_._1).toSeq
+    val discreteParams = paramDetails.zipWithIndex.filter(_._1._2 == "INT").map {
+      case ((_, _, min: Double, max: Double, _), index: Int) => index -> ((max - min).toInt + 1)
+    }.toMap
+    val ranges = paramDetails.map { case (_, _, min, max, _) => DoubleRange(min, max) }.toSeq
+
+    val transformMap = paramDetails.map(_._5).zipWithIndex.flatMap {
+      case (transform, index) => transform.map(trans => trans.toString match {
+        case LOG_TRANSFORM | SQRT_TRANSFORM => index -> trans.toString
+        case _ => throw new IllegalArgumentException("The transformation is not valid")
+      })
+    }.toMap
+
+    (tuningMode, hyperparameters, ranges, discreteParams, transformMap)
+  }
+
+  /**
+   * Helper method to read a file from HDFS as text.
+   *
+   * @param fs The Hadoop file system
+   * @param inputPath The HDFS input path from which to read the file
+   * @return The contents of the file as text
+   */
+  def readFile(fs: FileSystem, inputPath: String): String = {
```

Comment:
I think it is a good idea to keep these read and write code here. In the future, this will allow us to separate the module out.

Reply:
I am wondering if `readFile` and `writeJson` can be replaced by `IOUtils.readStringsFromHDFS` and `IOUtils.writeStringsToHDFS`. If so, why don't we use them in xgboost-ml?

Reply:
Let's do that change. I didn't know about this file while that initial version was created. Let's reuse existing files as much as we can.